lib/openstudio-standards/create_typical/space_type_information.rb

```diff
@@ -0,0 +1,160 @@
+  module CreateTypical
+    # @!group Space Type Information
+    # Assigns information like number of units, number of beds, number of students, etc. to spaces in the model. Some methods that add exterior lighting or elevators require this information
+
+    # Assign information like number of units, number of beds, number of students, etc. to spaces in the model.
+    #
+    # @param model [OpenStudio::Model::Model] OpenStudio model object
+    # @param assign_properties_to_spaces [Boolean] Assign properties to Space objects
+    # @return [Hash] A hash of space properties
+    def self.model_get_space_information(model, assign_properties_to_spaces: true)
+      spaces_hash = {}
+
+      model.getSpaces.sort.each do |space|
+        space_type = space.spaceType.is_initialized ? space.spaceType.get : nil
+        next if space_type.nil?
+
+        # skip space if not part of floor area
+        next unless space.partofTotalFloorArea
+
+        standards_building_type = space_type.standardsBuildingType.is_initialized ? space_type.standardsBuildingType.get : nil
+        standards_space_type = space_type.standardsSpaceType.is_initialized ? space_type.standardsSpaceType.get : nil
+        next if standards_space_type.nil?
+
+        effective_number_of_spaces = space.multiplier
+        floor_area = space.floorArea * space.multiplier
+        number_of_people = space.numberOfPeople * space.multiplier
+
+        # determine number of units
+        number_of_units = 0
+        case standards_space_type
+        when 'GuestRoom', 'GuestRm', 'GuestRmOcc', 'GuestRmUnOcc', 'GuestRoom123Occ', 'GuestRoom123Vac', 'GuestRoom4Occ', 'GuestRoom4Vac', 'PatientRoom', 'PatRoom'
+          average_unit_size = OpenStudio.convert(350.0, 'ft^2', 'm^2').get
+          average_unit_size = OpenStudio.convert(280.0, 'ft^2', 'm^2').get if standards_building_type == 'LargeHotel'
+          number_of_units = floor_area / average_unit_size
+        when 'Apartment', 'ResBedroom', 'ResLiving'
+          average_unit_size = OpenStudio.convert(950.0, 'ft^2', 'm^2').get
+          number_of_units = floor_area / average_unit_size
+        when 'Strip mall - type 1', 'Strip mall - type 2', 'Strip mall - type 3'
+          average_unit_size = OpenStudio.convert(2250.0, 'ft^2', 'm^2').get
+          number_of_units = floor_area / average_unit_size
+        end
+
+        # determine number of beds
+        number_of_beds = 0
+        case standards_space_type
+        when 'PatientRoom', 'PatRoom', 'HspSurgOutptLab', 'HspNursing', 'ICU_PatRm', 'ICU_Open'
+          number_of_beds = number_of_people
+        end
+
+        # determine number of students
+        number_of_students = 0
+        case standards_space_type
+        when 'Classroom'
+          typical_class_size = 20.0
+          number_of_students = number_of_people * ((typical_class_size - 1.0) / typical_class_size)
```

Comment:
I don't get this. Is the point to separate the number of people in a class into 1 teacher and the rest students? If so, why not just `number_of_students = number_of_people - 1`? With this, e.g. for a class with 5 people you get 4.75 students.

Reply:
I guess that is the case. Not sure why it was the old way. Will change.